### PR TITLE
Fix SetNoDraw in startFreeRoam

### PR DESCRIPTION
**Pull request by klen-list**
Entity still invisible after unselect in firstperson mode

https://user-images.githubusercontent.com/49125075/143343337-cc2ebec7-543e-444c-9e23-a3dff03aeb54.mp4

### CODE REVIEW

File: lua/fspectate/cl_init.lua [hunk 1]
```diff
@@ -380,11 +380,13 @@ end
 Start roaming free, rather than spectating a given player
 ---------------------------------------------------------------------------*/
 startFreeRoam = function()
-    if IsValid(specEnt) and specEnt:IsPlayer() then
-        roamPos = thirdperson and getThirdPersonPos(specEnt) or specEnt:GetShootPos()
+    if IsValid(specEnt) then
+        if specEnt:IsPlayer() then
+            roamPos = thirdperson and getThirdPersonPos(specEnt) or specEnt:GetShootPos()
+        else
+            roamPos = isSpectating and roamPos or LocalPlayer():GetShootPos()
```
Comment (FPtje) [inline lua/fspectate/cl_init.lua]: `roamPos` is no longer set when `specEnt` is not valid

**Review by FPtje — APPROVED** [review]
Oh wow, this bug must have been in there for years, and no one reported it. Even though FSpectate is in DarkRP by default. I guess that means it's not used widely 😞 

Either way, thanks a lot for the fix. I've noted one issue with it, and committed a fix.

**Comment by FlorianLeChat** [comment]
> Oh wow, this bug must have been in there for years, and no one reported it. Even though FSpectate is in DarkRP by default. I guess that means it's not used widely 😞

There is something really strange because I have never experienced this issue in all these years of using FSpectate, I can't even reproduce this bug on a recent version of gamemode. I've tested this in singleplayer and on a dedicated server but I can't see the props disappear.

**Comment by FPtje** [comment]
I could easily reproduce it by spectating a prop, going first person and then going into free roam again.

**Pull request merged**